llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -274,24 +397,41 @@ int PostPipeliner::mostUrgent() {
   return Best;
 }
 
-bool PostPipeliner::scheduleFirstIteration() {
+void PostPipeliner::resetSchedule(bool FullReset) {
+  Scoreboard.clear();
+  for (int K = 0; K < NTotalInstrs; K++) {
+    auto &N = Info[K];
+    N.reset(FullReset);
```

Comment:
Shouldn't a "full" reset also clear the "static" variables? Or do we always want to keep those because they never change, even when targeting a different II?

Reply:
They change on a new II, but then the entire array is cleared and reallocated.